mac_pw_pool/setup.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Setup and launch Cirrus-CI PW Pool node.  It must be called
+# with the env. var. `$POOLTOKEN` set.  It is assumed to be
+# running on a fresh AWS EC2 mac2.metal instance as `ec2-user`
+# or equivilent admin.  The instance must have both "metadata"
+# and "Allow tags in metadata" options enabled.
+
+set -eo pipefail
+
+GVPROXY_RELEASE_URL="https://github.com/containers/gvisor-tap-vsock/releases/latest/download/gvproxy-darwin"
+
+COMPLETION_FILE="$HOME/.setup.done"
+STARTED_FILE="$HOME/.setup.started"
+
+date -u -Iseconds >> "$STARTED_FILE"
+
+msg() { echo "##### ${1:-No message message provided}"; }
+die() { echo "ERROR: ${1:-No error message provided}"; exit 1; }
+
+[[ -n "$POOLTOKEN" ]] || \
+    die "Must be called with non-empty \$POOLTOKEN set."
+
+[[ ! -r "$COMPLETION_FILE" ]] || \
+    die "Appears setup script already ran at '$(cat $COMPLETION_FILE)'.  This script should not be called twice by automation."
+
+[[ "$USER" == "ec2-user" ]] || \
+    die "Expecting to execute as 'ec2-user'."
+
+# For whatever reason, when this script is run through ssh, the default
+# environment isn't loaded automatically.
+. /etc/profile
+
+msg "Configuring paths"
+grep -q homebrew /etc/paths || \
+    echo -e "/opt/homebrew/bin\n/opt/homebrew/opt/coreutils/libexec/gnubin\n$(cat /etc/paths)" \
+        | sudo tee /etc/paths > /dev/null
+
+msg "Installing podman-machine, testing, and CI deps. (~2m install time)"
+if [[ ! -x /usr/local/bin/gvproxy ]]; then
+    brew tap cfergeau/crc
+    brew install go go-md2man coreutils pstree vfkit cirruslabs/cli/cirrus
+
+    # Normally gvproxy is installed along with "podman" brew.  CI Tasks
+    # on this instance will be running from source builds, so gvproxy must
+    # be install from upstream release.
+    curl -sSLfO "$GVPROXY_RELEASE_URL"
+    sudo install -o root -g staff -m 0755 gvproxy-darwin /usr/local/bin/gvproxy
+    rm gvproxy-darwin
+fi
+
+msg "Adding/Configuring PW User"
+PWINST=$(curl -sSLf http://instance-data/latest/meta-data/tags/instance/Name)
```

Comment:
Assuming that [this](https://cloudinit.readthedocs.io/en/latest/explanation/instancedata.html) is what explains the magic `instance-data`, would you mind adding a comment here (and the other two) with a quick explanation and a link?

Reply:
Two things for this:
1. There is no cloud-init support on Macs ([they use this instead](https://github.com/aws/ec2-macos-init)). Remember Macs are each special unique rainbow snowflakes in a walled garden of unicorns, so Apple can ensure maximize profit and developer grief :rofl:
2. I dug this magic out of stackoverflow somewhere.  AWS docs can be hard on the eyes, they don't like to put everything in one place.  I think perhaps [either this](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html) or [this reference is maybe the most useful](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html).  Or maybe both?  I have no idea how they resolve "instance-data" to `169.254.169.254` it must be magic.